lib/rbs/prototype/rb.rb

```diff
@@ -586,6 +644,59 @@ def node_type(node, default: Types::Bases::Any.new(location: nil))
       def untyped
         @untyped ||= Types::Bases::Any.new(location: nil)
       end
+
+      def private
+        @private ||= AST::Members::Private.new(location: nil)
+      end
+
+      def public
+        @public ||= AST::Members::Public.new(location: nil)
+      end
+
+      def current_accessibility(decls, index = [0, decls.size - 1].max)
+        idx = decls.slice(0, index).rindex { |decl| decl == private || decl == public }
+        (idx && decls[idx]) || public
+      end
+
+      def remove_unnecessary_accessibility_methods!(decls)
+        current = public
+        idx = 0
+
+        loop do
+          decl = decls[idx] or break
+          if current == decl
+            decls.delete_at(idx)
+            next
+          end
+
+          if 0 < idx && is_accessibility?(decls[idx - 1]) && is_accessibility?(decl)
+            decls.delete_at(idx - 1)
+            idx -= 1
+            current = current_accessibility(decls, idx)
+            next
+          end
+
+          current = decl if is_accessibility?(decl)
+          idx += 1
+        end
+
+        decls.pop while decls.last && is_accessibility?(decls.last)
+      end
```

Comment:
`process` method may insert unnecessary accessibility methods. So it removes them when `process` method processed an entire class/module.


For example:


```ruby
# Ruby


class C
  private def foo() end
  private def bar() end
end
```

`prototype rb` will convert the Ruby code to the following RBS, if it works without `remove_unnecessary_accessibility_methods!`.

```ruby
# RBS

class C
  private
  def foo: () -> nil
  public # <- redundant

  private # <- redundant
  def bar: () -> nil
  public # <- redundant
```


So this method removes them.